app/actions/TrezorActions.js

```diff
@@ -794,13 +803,11 @@ export const updateFirmware = (path) => async (dispatch, getState) => {
 
   try {
     const rawFirmware = fs.readFileSync(path);
-
-    await deviceRun(dispatch, getState, async () => {
-      const res = await session.firmwareUpdate({
-        binary: rawFirmware.buffer
-      });
-      return res.payload;
-    });
+    const hexFirmware = rawToHex(rawFirmware);
+    // Ask main.development.js to send the firmware for us.
+    const errorStr = await ipcRenderer.invoke("upload-firmware", hexFirmware);
```

Comment:
It's best to send the path to the firmware file instead of the entire firmware contents to the ipc main thread.